hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/storage/HoodieConcatHandle.java

```diff
@@ -94,4 +107,34 @@ public void write(GenericRecord oldRecord) {
     }
     recordsWritten++;
   }
+
+  boolean needsUpdateLocation() {
+    return true;
+  }
+
+  @Override
+  public List<WriteStatus> close() {
+    try {
+      while (recordItr.hasNext()) {
+        HoodieRecord<T> record = recordItr.next();
```

Comment:
can we try to re-use code across merge handle and this. lines 119 to 133. If incase we need to make some fix in merge handle, we might miss to fix it here.

Reply:
Totally agree! I will refactor